app/controllers/api/v1/images_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Api::V1::ImagesController < ApplicationController
+    
+    def generate_presigned_url
+        object_key = "#{SecureRandom.uuid}"
+        presigner = Aws::S3::Presigner.new(client: S3_BUCKET.client)
+        presigned_url = presigner.presigned_url(:put_object, bucket: S3_BUCKET.name, key: object_key, expires_in: 3600)
+
+        render json: {presigned_url: presigned_url}
+    end
```

Comment:
this is a huge security risk here, since there is not validation anyone can upload anything to s3